### DISCUSSION TITLE
Annotations not working on pull_request

### DISCUSSION THREAD

**Discussion by st3fan** [community/community#25433] [Actions]
<p>I am working on an action that turns Xcode compile warnings and errors into annotations.</p>
<p>This Action works well when I use it in a workflow that is triggered by a push to the main branch. But when it is triggered by a <code>pull_request</code> the annotations do not show up. The workflow executes, I see the correct annotation in a debug log and the call to add the annotations to the check run is good. No annotations are shown though.</p>
<p>What do I miss here? Are PRs special wrt annotations and/or check runs?</p>

**Comment by st3fan** [comment]
<p>Here is the pull request:</p>
<aside class="onebox githubpullrequest">
  <header class="source">
      <a href="https://github.com/devbotsxyz/example-ios-hello/pull/4/checks?check_run_id=1245567820" target="_blank" rel="noopener nofollow ugc">github.com/devbotsxyz/example-ios-hello</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Pull Request">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 12 16" aria-hidden="true"><path d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/devbotsxyz/example-ios-hello/pull/4" target="_blank" rel="noopener nofollow ugc">Update AppDelegate.swift</a>
    </h4>

    <div class="branches">
      <code>devbotsxyz:main</code> ← <code>devbotsxyz:st3fan-patch-4</code>
    </div>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2020-10-13" data-time="03:43:51" data-timezone="UTC">03:43AM - 13 Oct 20 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/st3fan" target="_blank" rel="noopener nofollow ugc">
          <img alt="st3fan" src="https://user-images.githubusercontent.com/28052/181091098-d15ddde0-b884-495d-96ef-7b8f5967b0ad.jpeg" class="onebox-avatar-inline" width="20" height="20">
          st3fan
        </a>
      </div>

      <div class="lines" title="1 commits changed 1 files with 2 additions and 0 deletions">
        <a href="https://github.com/devbotsxyz/example-ios-hello/pull/4/files" target="_blank" rel="noopener nofollow ugc">
          <span class="added">+2</span>
          <span class="removed">-0</span>
        </a>
      </div>
    </div>

  </div>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>And this is a regular commit to main:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/devbotsxyz/example-ios-hello/actions/runs/303597136" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/28052/181091099-9ab041de-a2ca-4eeb-8cb8-5fd66047f1a9.png" class="thumbnail onebox-avatar" width="400" height="400">

<h3><a href="https://github.com/devbotsxyz/example-ios-hello/actions/runs/303597136" target="_blank" rel="noopener nofollow ugc">devbotsxyz/example-ios-hello</a></h3>

<p>Example iOS Hello. Contribute to devbotsxyz/example-ios-hello development by creating an account on GitHub.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>The first has no annotations and no sign of the check. The latter works as I would expect.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/st3fan">@st3fan</a>,</p>
<p>In the docs about the <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/checks#create-a-check-run" rel="noopener nofollow ugc">Checks API</a>, I noticed the following description:</p>
<blockquote>
<p><strong>Note:</strong>  The Checks API only looks for pushes in the repository where the check suite or check run were created. Pushes to a branch in a forked repository are not detected and return an empty <strong>pull_requests</strong> array.</p>
</blockquote>
<p>When the workflow runs on pull request, the commit SHA (<strong>github.sha</strong>) is the last merge commit on the PR merge branch (<strong>refs/pull/:prNumber/merge</strong>). It is not a commit that has been pushed into the repository.<br>
At this time, you should specify the latest commit on the head branch (source branch) of the PR as the <strong>head_sha</strong> of the API. To get the head SHA, you can use the expression <strong>${{ github.event.pull_request.head.sha }}</strong>.</p>

**Comment by st3fan** [comment, marked as answer]
<p>I’ll close this thread because I think in the end I was simply testing the wrong thing. My mistake.</p>
